pkg/authorization/authorizer/scope/converter.go

```diff
@@ -0,0 +1,140 @@
+package scope
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/authorization/rulevalidation"
+	userapi "github.com/openshift/origin/pkg/user/api"
+)
+
+func ScopesToRules(scopes []string, namespace string, clusterPolicyGetter rulevalidation.ClusterPolicyGetter) ([]authorizationapi.PolicyRule, error) {
+	rules := append([]authorizationapi.PolicyRule{}, discoveryRules...)
+
+	errors := []error{}
+	for _, scope := range scopes {
+		for prefix, evaluator := range scopeEvaluators {
+			if strings.HasPrefix(scope, prefix) {
+				currRules, err := evaluator(scope, namespace, clusterPolicyGetter)
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+
+				rules = append(rules, currRules...)
+			}
+		}
+	}
+
+	return rules, kutilerrors.NewAggregate(errors)
+}
+
+var discoveryRules = []authorizationapi.PolicyRule{
+	{
+		Verbs: sets.NewString("get"),
+		NonResourceURLs: sets.NewString(
+			// API discovery/negotiation
+			"/api", "/api/*",
+			"/apis", "/apis/*",
+			"/oapi", "/oapi/*",
+		),
+	},
+}
+
+const (
+	SpecialIndicator       = "special:"
+	ClusterRoleIndicator   = "clusterrole:"
+	ClusterWideIndicator   = "clusterwide:"
+	NamespaceWideIndicator = "namespace:"
+)
+
+// scopeEvaluator takes a scope and returns the rules that express it
+type scopeEvaluator func(scope, namespace string, clusterPolicyGetter rulevalidation.ClusterPolicyGetter) ([]authorizationapi.PolicyRule, error)
+
+// scopeEvaluators map prefixes to a function that handles that prefix
+var scopeEvaluators = map[string]scopeEvaluator{
+	SpecialIndicator:     specialEvaluator,
+	ClusterRoleIndicator: clusterRoleEvaluator,
+}
+
+// scopes are in the format
+// <indicator><indicator choice>
+// we have the following formats:
+// clusterwide:<comma-delimitted verbs>:<comma-delimitted resources>
+// namespace:<namespace name>:<comma-delimitted verbs>:<comma-delimitted resources>
```

Comment:
> Needs to be updated? Also, comma-delimited

More of a TODO.  You said these were secondary objectives for kind of crazy people.